### DISCUSSION TITLE
Flux Deployment with HPA

### DISCUSSION THREAD

**Discussion by dipti-pai** [fluxcd/flux2#4987] [Q&A]
Scenario: 

There is a known behavior when Deployments with HPA are used with Flux where Flux reconciliation overwrites the HPA replica settings in deployment. The workaround suggested by HPA and Flux in this [FAQ](https://fluxcd.io/flux/faq/#what-is-a-kustomization-reconciliation) is to remove the .spec.replicas from the Deployment spec. 

Issue:

For existing deployments in production, removing `.spec.replicas` results in a temporary degradation where number of replicas is reduced to one. The [recommendation in HPA docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#migrating-deployments-and-statefulsets-to-horizontal-autoscaling) to workaround this degradation is to transfer ownership when using server-side apply. While the suggested workaround is possible when there are manual cluster operators with a few clusters, it is not feasible with a large fleet of clusters with thousands of deployments deployed using Flux. 

Question: 

Is there an alternate way to address this when HPA is used with deployments? Is it feasible to detect the conflict and remove the replica configuration when the controller applies the diff ?

**Comment by stefanprodan** [comment]
There is an issue for this that if implemented would solve so many other issues than with HPA. See https://github.com/fluxcd/pkg/issues/696
